src/commands/synthetics/deploy-tests-lib.ts

```diff
@@ -0,0 +1,93 @@
+import {getCommonAppBaseURL} from '../../helpers/app'
+
+import {APIHelper, EndpointError, formatBackendErrors, getApiHelper} from './api'
+import {DeployTestsCommandConfig, LocalTriggerConfig, MainReporter, ServerTest, TestConfig} from './interfaces'
+import {findUniqueLocalTestDefinition} from './local-test-definition'
+import {getTestConfigs} from './test'
+import {isLocalTriggerConfig} from './utils/internal'
+
+export const deployTests = async (reporter: MainReporter, config: DeployTestsCommandConfig): Promise<void> => {
+  const api = getApiHelper(config)
+
+  reporter.log('Deploying tests...\n\n')
+
+  const testConfigFromFile: TestConfig = {
+    tests: await getTestConfigs(config, reporter),
+  }
+
+  if (config.publicIds.length > 0) {
+    testConfigFromFile.tests = testConfigFromFile.tests.filter(
+      (test) =>
+        isLocalTriggerConfig(test) &&
+        test.local_test_definition.public_id &&
+        config.publicIds.includes(test.local_test_definition.public_id)
+    )
+  }
```

Comment:
Maybe let's keep the same behavior as the `run-tests` command: if no public id is provided, it deploy everything?

Reply:
When you say "it deploys everything", do you mean everything in all the `.synthetics.json` files, or everything in the single test file?

I wanted to align with the other PRs we made about LTDs to only support a single test file for now

Reply:
I think we should align on the run-test command, and do the action on everything. If `-f` is provided, then the single file is used, otherwise, do the action on all the files that are available.